src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -524,7 +524,7 @@ void populateDiagnosticsAndClear(ArrayBuilder<TypeParameterDiagnosticInfo> build
 
             var diagnosticsBuilder = ArrayBuilder<TypeParameterDiagnosticInfo>.GetInstance();
             ArrayBuilder<TypeParameterDiagnosticInfo> useSiteDiagnosticsBuilder = null;
-            var result = !typeSyntax.HasErrors && CheckTypeConstraints(type, conversions, includeNullability, currentCompilation, diagnosticsBuilder, nullabilityDiagnosticsBuilderOpt: diagnosticsBuilder, ref useSiteDiagnosticsBuilder);
+            var result = !typeSyntax.HasErrors && CheckTypeConstraints(type, conversions, currentCompilation, diagnosticsBuilder, nullabilityDiagnosticsBuilderOpt: includeNullability ? diagnosticsBuilder : null, ref useSiteDiagnosticsBuilder);
```

Comment:
nit: line is 244 characters, consider wrapping new lines for the arguments

Reply:
I'll leave these long lines as-is since the issues were not introduced in this change.

---
In reply to: [405767579](https://github.com/dotnet/roslyn/pull/43094#discussion_r405767579) [](ancestors = 405767579)